src/control/server/README.md

```diff
@@ -1,431 +1,199 @@
+over dRPC channel and handled by the [mgmt](/src/mgmt/srv.c) module.
 
-The DAOS data plane utilises two forms of non-volatile storage, storage class memory (SCM) in the form of persistent memory modules and NVMe in the form of high-performance SSDs.
-
-The DAOS control plane provides capability to provision and manage the non-volatile storage including the allocation of resources to data plane instances.
+Management service RPC handler code is contained in
+[`mgmt_svc.go`](/src/control/server/mgmt_svc.go).
+
+### Fanout
+
+Some control service RPC handlers will trigger fanout to multiple
+remote harnesses over gRPC, in order to send these fanout requests
+the `mgmtClient` is used as defined in
+[`mgmt_client.go`](/src/control/server/mgmt_client.go).
+
+An example of a management tool command that executes gRPC fanout
+over multiple remote harnesses is `dmg system query`, the server side
+handler for which is `SystemQuery` in
+[`ctl_system.go`](/src/control/server/ctl_system.go) which issues
+requests to remote harnesses using the `HarnessClient` abstraction in
+[`ctl_harness.go`](/src/control/server/ctl_harness.go) which in turn
+sends requests with the `mgmtClient` as described above.
+
+### Storage
+
+Storage related RPCs, whose handlers are defined in
+[`ctl_storage*.go`](/src/control/server/ctl_storage.go)
+delegate operations to backend providers encapsulated in the `bdev`
+and `scm`
+[storage subpackages](/src/control/server/storage/).
+
+## Bootstrapping and DAOS system membership
 
-Storage format is required after other storage management operations have been performed as a precursor to bringing up the DAOS data plane:
+When starting a data-plane instance, we look at the superblock to
+determine whether it should be a MS (management service) replica.
+The `daos_server.yml`'s `access_points` parameter is used (only
+during format) to determine whether an instance is to be a MS replica
+or not.
+
+When the starting instance is identified as an MS replica, it
+performs bootstrap and starts.  If the DAOS system has only one
+replica (as specified by `access_points` parameter), the host of the
+bootstrapped instance is now the MS leader.  Whereas if there are
+multiple replicas, elections will happen in the background and
+eventually a leader will be elected.
+
+When the starting instance is not identified as an MS replica, the
+instance's host calls Join on MgmtSvcClient over gRPC including the
+instance's host ControlAddress (address that the gRPC server is
+listening on) in the request addressed to the MS leader.
+
+MgmtSvc running on the MS leader handles the Join request received by
+gRPC server and forwards request over dRPC to the MS leader instance.
+If the Join request is successful then the MS leader MgmtSvc records
+the address contained in the request as a new system member.
+
+## Storage Management
+
+Operations on NVMe SSD devices are performed using
+[go-spdk bindings](https://github.com/daos-stack/go-spdk)
```

Comment:
Seems strange to link to this repo, given that it's defunct and will likely be removed. Probably better to link to lib/spdk as you did with lib/ipmctl, no?

Reply:
yes thanks for the catch